src/util/domains/check-transfer.ts

```diff
@@ -0,0 +1,11 @@
+import Client from '../client';
+
+type Response = {
+  transferrable: boolean;
+  status: string;
+  reason: string;
+};
+
+export default async function checkTransfer(client: Client, name: string) {
+  return await client.fetch<Response>(`/v4/domains/${name}/transfer`);
```

Comment:
If we are not using the reason anywhere it might be better to make this an async predicate function